src/formatters/utils/sortResults.ts

```diff
@@ -1,13 +1,79 @@
 import { IRuleResult } from '../../types';
 
-export const sortResults = (results: IRuleResult[]) => {
-  return [...results].sort((resultA, resultB) => {
-    const diff = resultA.range.start.line - resultB.range.start.line;
+const compareCode = (one: string | number | undefined, another: string | number | undefined): number => {
+  if (one === void 0 && another === void 0) {
+    return 0;
+  }
+
+  if (one === void 0) {
+    return -1;
+  }
+
+  if (another === void 0) {
+    return 1;
+  }
+
+  return one.toString().localeCompare(another.toString(), undefined, { numeric: true });
+};
+
+const compareSource = (one: string | undefined, another: string | undefined): number => {
+  if (one === void 0 && another === void 0) {
+    return 0;
+  }
+
+  if (one === void 0) {
+    return -1;
+  }
 
-    if (diff === 0) {
-      return resultA.range.start.character - resultB.range.start.character;
-    }
+  if (another === void 0) {
+    return 1;
+  }
+
+  return one.localeCompare(another);
+};
+
+const normalize = (value: number): -1 | 0 | 1 => {
+  if (value < 0) {
+    return -1;
+  }
+
+  if (value > 0) {
+    return 1;
+  }
+
+  return 0;
+};
 
-    return diff;
-  });
+export const resultsComparer = (resultA: IRuleResult, resultB: IRuleResult): -1 | 0 | 1 => {
```

Comment:
Now that I think of it, I believe we could name it in the way other functions are named in this file.
`compareResults` What do you think?

Reply:
Fixed